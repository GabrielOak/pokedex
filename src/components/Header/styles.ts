import styled from 'styled-components/native';

export const Wrapper = styled.View`
  height: 150px;
  width: 100%;
  background-color: #403f3f;
  border-bottom-left-radius: 30px;
  border-bottom-right-radius: 30px;
  justify-content: center;
  padding: 20px;
`;

export const PokeballImage = styled.Image`
  position: absolute;
  width: 120px;
  height: 120px;
  align-self: flex-end;
`;

export const ProfileImage = styled.Image`
  width: 60px;
  height: 60px;
  background-color: white;
  border-radius: 10px;
  margin-right: 8px;
  margin-top: 6px;
`;

export const InfoContainer = styled.View`
  margin-top: 20px;
`;

export const HeaderContainer = styled.View`
  flex-direction: row;
  justify-content: space-between;
`;

export const Title = styled.Text`
  color: white;
  font-weight: 600;
  font-size: 18px;
`;

export const SubTitle = styled.Text`
  color: white;
  font-weight: 500;
  font-size: 16px;
`;

export const TitleBold = styled.Text`
  color: white;
  font-weight: bold;
  font-size: 18px;
`;
