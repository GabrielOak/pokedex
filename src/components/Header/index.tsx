import React from 'react';
import Pokeball from '../../assets/pokeball.png';
import Ash from '../../assets/ash.png';

import * as S from './styles';

const Header = () => {
  return (
    <S.Wrapper>
      <S.PokeballImage source={Pokeball} />
      <S.HeaderContainer>
        <S.InfoContainer>
          <S.Title>
            OlÃ¡, <S.TitleBold>Ash Ketchum</S.TitleBold>
          </S.Title>
          <S.SubTitle>Bem vindo! ğŸ˜„</S.SubTitle>
        </S.InfoContainer>
        <S.ProfileImage source={Ash} />
      </S.HeaderContainer>
    </S.Wrapper>
  );
};
export default Header;
