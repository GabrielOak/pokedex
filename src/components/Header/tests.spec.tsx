import React from 'react';
import {render} from '@testing-library/react-native';
import Header from '.';

describe('<Header />', () => {
  it('should render correcly', () => {
    const {getByText, toJSON} = render(<Header />);

    expect(getByText(/bem vindo/i)).toBeTruthy();
    expect(getByText(/ash ketchum/i)).toBeTruthy();
    expect(getByText(/ðŸ˜„/i)).toBeTruthy();

    expect(toJSON()).toMatchSnapshot();
  });
});
