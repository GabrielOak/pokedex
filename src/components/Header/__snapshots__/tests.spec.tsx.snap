// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render correcly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#403f3f",
        "borderBottomLeftRadius": 30,
        "borderBottomRightRadius": 30,
        "height": 150,
        "justifyContent": "center",
        "paddingBottom": 20,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 20,
        "width": "100%",
      },
    ]
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/pokeball.png",
      }
    }
    style={
      Array [
        Object {
          "alignSelf": "flex-end",
          "height": 120,
          "position": "absolute",
          "width": 120,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "marginTop": 20,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 18,
              "fontWeight": "600",
            },
          ]
        }
      >
        OlÃ¡, 
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 18,
                "fontWeight": "bold",
              },
            ]
          }
        >
          Ash Ketchum
        </Text>
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 16,
              "fontWeight": "500",
            },
          ]
        }
      >
        Bem vindo! ðŸ˜„
      </Text>
    </View>
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/ash.png",
        }
      }
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderRadius": 10,
            "height": 60,
            "marginRight": 8,
            "marginTop": 6,
            "width": 60,
          },
        ]
      }
    />
  </View>
</View>
`;
